make/prepare

```diff
@@ -23,13 +23,14 @@ def validate(conf, args):
     if args.ha_mode:
         db_host = rcp.get("configuration", "db_host")
         if db_host == "mysql":
-            raise Exception("Error: In HA mode, db_host in harbor.cfg needs to point to an external DB address")
-        registry_config_path = os.path.join(templates_dir,"registry","config_ha.yml")
-        if check_storage_config(registry_config_path):
-            raise Exception("Error: In HA model shared storage configuration is required registry, refer HA installation guide for detail.")
+            raise Exception("Error: In HA mode, db_host in harbor.cfg needs to point to an external DB address.")
+        registry_storage_provider = rcp.get("configuration",
+                                            "registry_storage_provider").strip()
+        if registry_storage_provider == 'filesystem':
+            raise Exception("Error: In HA mode, shared storage configuration in harbor.cfg is required. Refer to HA installation guide for details.")
```

Comment:
User can still use file system. as Ceph, NFS.  they are also filesystem.

Reply:
'filesystem' is a type supported by Registry per https://docs.docker.com/registry/configuration/#storage. For NFS, it's already mounted on the OS, so we don't need to distinguish it.

Reply:
Get your point now. Since we do not know whether it's local file system or NFS when the storage provider is 'filesystem', then no need to check it here? @yixingjia is this error tip in the HA installation guide too ? I guess the old code  prints this error message to the users to guide them configure the correct registry storage provider.

Reply:
The old message is just to make sure User understanding what they are doing when setup HA. Since by default we use local fileysystem.  But the we don't know the backend support if they use local storage then there will be issue in HA setup.

Reply:
After discussion with Yixing, we will ask the user a question to confirm shared storage is configured for Registry:
Is the Harbor Docker Registry configured to use shared storage (e.g. NFS, S3, GCS, etc.)? [yes/no]: